core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -165,20 +175,55 @@ private void setTelemetryModules(ApplicationInsightsXmlConfiguration appConfigur
     }
 
     /**
-     * Setting an instrumentation key
-     * @param instrumentationKey The instrumentation key found in the configuration.
+     * Setting an instrumentation key:
+     * First we try the JVM option '-DAPPLICATION_INSIGHTS_IKEY=i_key'
```

Comment:
We shouldn't check for a JVM option but for a **[system property](https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html)**